pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/PrimitiveWrapperInstantiationRule.java

```diff
@@ -55,34 +48,27 @@ public Object visit(ASTAllocationExpression node, Object data) {
      * Finds calls of "Boolean.valueOf".
      */
     @Override
-    public Object visit(ASTPrimaryExpression node, Object data) {
-        if (!TypeTestUtil.isA(Boolean.class, node)) {
-            return data;
-        }
-
-        if (node.getNumChildren() >= 2 && node.getChild(0).getNumChildren() > 0
-                && node.getChild(0).getChild(0) instanceof ASTName
-                && node.getChild(0).getChild(0).hasImageEqualTo("Boolean.valueOf")) {
-            ASTPrimarySuffix suffix = (ASTPrimarySuffix) node.getChild(1);
-            checkArguments(suffix.getFirstChildOfType(ASTArguments.class), node, data);
+    public Object visit(ASTMethodCall node, Object data) {
+        if (InvocationMatcher.parse("java.lang.Boolean#valueOf(_)").matchesCall(node)) {
```

Comment:
`InvocationMatcher` instances are immutable, aren't them? we should be able to move `InvocationMatcher.parse("java.lang.Boolean#valueOf(_)")` to a static final field in that case and avoid re-parsing on each method call of each file.

Reply:
also thinking… this could be added as a rule at https://github.com/pmd/build-tools/blob/master/src/main/resources/net/sourceforge/pmd/pmd-dogfood-config.xml

Reply:
:+1:

Reply:
Done: https://github.com/pmd/build-tools/commit/388ab2d78816c833c097ac9cb90ac7697aa884a6